src/main/kotlin/net/lumalyte/lg/infrastructure/services/ARMIntegrationService.kt

```
@@ -5,81 +5,72 @@ import org.bukkit.Location
-            val region = armAdapter?.getRegion(location)
-            region != null // If a region exists at this location, it's a shop region
+            val lookupClass = Class.forName("net.badgersmc.em.api.ShopGuildLookup")
+            val lookup = Bukkit.getServicesManager().load(lookupClass)
+
+            if (lookup != null) {
+                val isShopContainerMethod = lookupClass.getMethod("isShopContainer", Location::class.java)
+                isShopContainerMethod.invoke(lookup, location) as? Boolean ?: false
+            } else {
+                false
+            }
         } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)